sql3/planner/compileselect.go

```diff
@@ -129,9 +128,49 @@ func (p *ExecutionPlanner) compileSelectStatement(stmt *parser.SelectStatement,
 		}
 	}
 
-	// do we have straight projection or a group by?
+	// compile order by and generate a list of ordering expressions
+	orderByExprs := make([]*OrderByExpression, 0)
+	nonReferenceOrderByExpressions := make([]types.PlanExpression, 0)
+	if len(stmt.OrderingTerms) > 0 {
+		for _, ot := range stmt.OrderingTerms {
+			// compile the ordering term
+			expr, err := p.compileOrderingTermExpr(ot.X, projections, stmt.Source)
+			if err != nil {
+				return nil, err
+			}
+
+			f := &OrderByExpression{
+				Expr: expr,
+			}
+			f.Order = orderByAsc
+			if ot.Desc.IsValid() {
+				f.Order = orderByDesc
+			}
+			orderByExprs = append(orderByExprs, f)
+		}
+
+		// if the expression is just references, we
+		// can put the sort directly after the source
+		for _, oe := range orderByExprs {
+			_, ok := oe.Expr.(*qualifiedRefPlanExpression)
```

Comment:
I'm not sure I understand this. So, what I'm inferring is that if we compile a thing, and it is actually already present in the plan, we always get a `qualifiedRefPlanExpression`, but we can't get that if it's not already present in the plan? I guess the name sort of implies that.

Reply:
I don't understand the question?